pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,430 @@
+	PrevForceMatchingSignature *string        `db:"PREV_FORCE_MATCHING_SIGNATURE"`
+	PrevSQLPlanHashValue       *uint64        `db:"PREV_SQL_PLAN_HASH_VALUE"`
+	PrevSQLExecStart           sql.NullString `db:"PREV_SQL_EXEC_START"`
+	PrevSQLAddress             sql.NullString `db:"PREV_SQL_ADDRESS"`
+	Module                     sql.NullString `db:"MODULE"`
+	Action                     sql.NullString `db:"ACTION"`
+	ClientInfo                 sql.NullString `db:"CLIENT_INFO"`
+	LogonTime                  sql.NullString `db:"LOGON_TIME"`
+	ClientIdentifier           sql.NullString `db:"CLIENT_IDENTIFIER"`
+	OpFlags                    uint64         `db:"OP_FLAGS"`
+	BlockingInstance           *uint64        `db:"BLOCKING_INSTANCE"`
+	BlockingSession            *uint64        `db:"BLOCKING_SESSION"`
+	FinalBlockingInstance      *uint64        `db:"FINAL_BLOCKING_INSTANCE"`
+	FinalBlockingSession       *uint64        `db:"FINAL_BLOCKING_SESSION"`
+	WaitEvent                  sql.NullString `db:"EVENT"`
+	WaitEventGroup             sql.NullString `db:"WAIT_CLASS"`
+	WaitTimeMicro              *uint64        `db:"WAIT_TIME_MICRO"`
+	Statement                  sql.NullString `db:"SQL_FULLTEXT"`
+	PrevSQLFullText            sql.NullString `db:"PREV_SQL_FULLTEXT"`
+	PdbName                    sql.NullString `db:"PDB_NAME"`
+	CommandName                sql.NullString `db:"COMMAND_NAME"`
+}
+
+func (c *Check) getSQLRow(SQLID sql.NullString, forceMatchingSignature *string, SQLPlanHashValue *uint64, SQLExecStart sql.NullString) (OracleSQLRow, error) {
+	SQLRow := OracleSQLRow{}
+	if SQLID.Valid {
+		SQLRow.SQLID = SQLID.String
+		c.statementsFilter.SQLIDs[SQLID.String] = 1
+	} else {
+		SQLRow.SQLID = ""
+		return SQLRow, nil
+	}
+	if forceMatchingSignature != nil {
+		forceMatchingSignatureUint64, err := strconv.ParseUint(*forceMatchingSignature, 10, 64)
+		if err != nil {
+			return SQLRow, fmt.Errorf("failed converting force_matching_signature to uint64 %w", err)
+		}
+		SQLRow.ForceMatchingSignature = forceMatchingSignatureUint64
+		c.statementsFilter.ForceMatchingSignatures[*forceMatchingSignature] = 1
+	} else {
+		SQLRow.ForceMatchingSignature = 0
+	}
+	if SQLPlanHashValue != nil {
+		SQLRow.SQLPlanHashValue = *SQLPlanHashValue
+	}
+	if SQLExecStart.Valid {
+		SQLRow.SQLExecStart = SQLExecStart.String
+	}
+	return SQLRow, nil
+}
+
+func (c *Check) SampleSession() error {
+	start := time.Now()
+
+	if c.statementsFilter.SQLIDs == nil {
+		c.statementsFilter.SQLIDs = make(map[string]int)
+	}
+	if c.statementsFilter.ForceMatchingSignatures == nil {
+		c.statementsFilter.ForceMatchingSignatures = make(map[string]int)
+	}
+
+	var sessionRows []OracleActivityRow
+	sessionSamples := []OracleActivityRowDB{}
+	err := c.db.Select(&sessionSamples, ACTIVITY_QUERY)
+
+	if err != nil {
+		return fmt.Errorf("failed to collect session sampling activity: %w", err)
+	}
+
+	o := obfuscate.NewObfuscator(obfuscate.Config{SQL: c.config.ObfuscatorOptions})
```

Comment:
Looks like the Obfuscator sends statsd metrics. Is it safe to use here?

Reply:
We're using the obfuscator in [all other database integrations](https://github.com/DataDog/integrations-core/blob/a3ec3257adb01fe1405194f179fedb2c839b5869/datadog_checks_base/datadog_checks/base/utils/db/utils.py#L190). What are exactly safety concerns related to statsd metrics?

Reply:
Can obfuscator be safely instantiated multiple times? Python checks use a singleton instance as far as I can see.

Where does it sends the metrics? Also, multiple instances sending the same gauge without extra tags will likely overwrite each other.

Reply:
> Can obfuscator be safely instantiated multiple times? 

According to @alexandre-normand - yes (https://dd.slack.com/archives/C04HV5BPBU2/p1677255976977959?thread_ts=1677180393.067479&cid=C04HV5BPBU2):

> from a quick scan, I think the concurrency limitation is not because of some global thing but more about the state that's kept in an instance of the obfuscator. So as long as each oracle instance would have its own instance of the obfuscator, it looks like we'd be fine

> Where does it sends the metrics? 
I don't think the obfuscator sends any metrics in my case, because I'm not initialising Statsd field in the `Config` structure (https://github.com/DataDog/datadog-agent/blob/main/pkg/obfuscate/obfuscate.go#L86)